packages/compass/src/app/index.tsx

```diff
@@ -253,6 +259,14 @@ const Application = View.extend({
       this.queryByHook('layout-container')
     );
 
+    if (!isSecretStorageAvailable) {
+      openToast('secret-storage-not-available', {
+        variant: 'warning',
+        title:
+          'Compass cannot access credential storage. You can still connect, but please note that passwords will not be saved.',
```

Comment:
I was thinking about it, but still have doubts. On one hand, the toast might seem annoying, on the other hand, if they disabled this with the setting, they might forget about it over time. My initial idea with the warning pillow on the connect form came because of the extra reason, to always display this information to users, similarly to how we inform them in the case of CSFLE, that credentials will be stored on disk. With the current implementation, we show the toast on each Compass launch, not on disconnect.

Reply:
> My initial idea with the warning pillow on the connect form came because of the extra reason, to always display this information to users, similarly to how we inform them in the case of CSFLE, that credentials will be stored on disk.

I like that initial approach, it's less intrusive to someone who knows what they're doing on a custom OS setup. 
Looking at the slack thread: https://mongodb.slack.com/archives/GDTJXPHD0/p1721654142590129 
Sounds like Julia likes it. I'm cool with either approach, although I do think if we do the toast we should think about if how folks will want to hide it.

Reply:
Agreed in Slack do not introduce a settign for now, but add a telemetry event to track the number of occurrences.